lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -9,10 +9,261 @@
+            bool of success/failure
+        """
+        assert body is not None
+
+        api_response = self.api.create_application_from_template_using_post(
+            body)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return api_response.object_uuid
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+    def delete(self, uuid):
+        """ Deletes an application specified by UUID, returns bool of success/failure
+        """
+        assert uuid is not None
+
+        api_response = self.api.delete_application_using_delete(uuid)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return True
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+    def power_action_on_instance_by_uuid(self, uuid, power_action):
+        """ Performs a specified power operation on an application instance
+            specified by UUID, returns bool of success/failure
+        """
+        power_action_dict = {
+            ACTION_started: self.api.start_application_using_put,
+            ACTION_stopped: self.api.stop_application_using_put,
+            ACTION_restarted: self.api.restart_application_using_put,
+            ACTION_shutdown: self.api.shutdown_application_using_put,
+            ACTION_force_restarted: self.api.force_restart_application_using_put,
+            ACTION_paused: self.api.pause_application_using_put,
+            ACTION_resumed: self.api.resume_application_using_put,
+            ACTION_absent: self.api.delete_application_using_delete
+        }
+        assert uuid is not None
+        assert power_action in power_action_dict
+
+        api_response = power_action_dict[power_action](uuid)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return True
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+
+class VlanResource(object):
+    def __init__(self, api_client):
+        self.api = tacp.VlansApi(api_client)
+        self.api_client = api_client
+
+    def get_all(self):
```

Comment:
I dont see the reason why you would need all vlans. Even if you did, you could make this a little bit more robust by

1. change the method name to `filter` and catch all named arguments `**kws)
2. make a string out of the `**kws`, chain them by `;` and send this string as `filters` to the api

Reply:
I'm leaving the name get_all, but passing **kwargs to the filters parameter directly since the function takes a python list as the filter parameter.